entrypoint.py

```diff
@@ -321,29 +333,35 @@ def _copy_java_validators():
         action="store_true",
     )
     args = parser.parse_args()
-
-    if args.environs or args.dump_config:
-        _show_environ(Path(args.config_dir), dump_config=args.dump_config)
-    tee = subprocess.Popen(
-        ["/usr/bin/tee", "super-sast.log"], shell=False, stdin=subprocess.PIPE
-    )
-    # Cause tee's stdin to get a copy of our stdin/stdout (as well as that
-    # of any child processes we spawn)
-    os.dup2(tee.stdin.fileno(), stdout.fileno())
-    os.dup2(tee.stdin.fileno(), stderr.fileno())
-    sast_status = {}
-
-    _copy_java_validators()
-    run_all = os.environ.get("RUN_ALL_TOOLS", "true").lower() == "true"
-    for tool, command in TOOLS_MAP.items():
-        status = run_sast(
-            tool,
-            command,
-            os.environ.copy(),
-            config_dir=Path(args.config_dir),
-            log_file=stdout,
-            run_all=run_all,
+    try:
+        if args.environs or args.dump_config:
+            _show_environ(Path(args.config_dir), dump_config=args.dump_config)
+        tee = subprocess.Popen(
+            ["/usr/bin/tee", "super-sast.log"], shell=False, stdin=subprocess.PIPE
+        )
+        # Cause tee's stdin to get a copy of our stdin/stdout (as well as that
+        # of any child processes we spawn)
+        os.dup2(tee.stdin.fileno(), stdout.fileno())
+        os.dup2(tee.stdin.fileno(), stderr.fileno())
+        sast_status = {}
+
+        _copy_java_validators()
+        run_all = os.environ.get("RUN_ALL_TOOLS", "true").lower() == "true"
+        for tool, command in TOOLS_MAP.items():
+            status = run_sast(
+                tool,
+                command,
+                os.environ.copy(),
+                config_dir=Path(args.config_dir),
+                log_file=stdout,
+                run_all=run_all,
+            )
+            sast_status[tool] = status
+        log.info("All tools finished")
+        log.info(sast_status)
+    except Exception as e:
+        log.error(
+            "An exception occurred while running SAST, context information below\n"
+            f" - UID: {os.getuid()}\n - GID: {os.getgid()}\n"
+            f" - Error: {type(e).__name__} -> {e}"
         )
```

Comment:
```suggestion
        )
        raise e
```

Since the software is in beta, I prefer to reraise exceptions and fail fast, returning a non-zero status.